### ISSUE TITLE
[FUNCTIONALITY]: Modal - Focus MUST be set correctly on the window for all screen readers

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#9231]
# [508-defect-1 :exclamation: Launchblocker](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1)

**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️  Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

While testing a modal window for the SSOe team with iOS Voiceover on Safari, I noticed our [Modal component](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/blob/master/packages/formation-react/src/components/modal/Modal.jsx) is not setting focus correctly on the close button. The yellow halo is being set, but the iOS virtual cursor is not. While this issue is limited to the iOS Safari + VoiceOver combination, it is a big issue because a large portion of our mobile traffic is iOS Safari.

## Related Issues
* https://github.com/department-of-veterans-affairs/va.gov-team/issues/1449

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Acceptance Criteria

- [ ] iOS VoiceOver + Safari properly moves the virtual cursor to the close button
- [ ] Swiping right highlights the heading and not the content "behind" the modal window

## Environment

* iOS latest
* Safari
* VoiceOver on iPhone 8
* staging.va.gov

## Steps to Recreate

1. Start VoiceOver on iOS. It is located in the Settings App > Accessibility > VoiceOver
1. Clear you iOS Safari cache to ensure the first-time modal fires
1. Browse to staging.va.gov on Safari
1. Log in with any relevant test credentials
1. Wait for the modal to appear
1. Swipe right and verify the black virtual cursor outline is NOT on the modal heading

## WCAG or Vendor Guidance (optional)

* [Focus Order: Understanding SC 2.4.3](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-order.html)

**Comment by 1Copenut** [comment]
Issue moved to [department-of-veterans-affairs/vets-design-system-documentation #226](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/226) via [**ZenHub**](https://www.zenhub.com/)
